panic on invalid `sourceValue`

### What is the environment (Minikube, Openshift)?
CodeReady Containers version: 1.26.0+31f06c09
OpenShift version: 4.7.8 (embedded in executable)

### What is the SBO version used?
v0.7.1

### What are the steps to reproduce this issue?
  1. add an annotation to `Service` with invalid  `sourceValue`
   
```
service.binding/java-maven_port: path={.spec.ports},elementType=sliceOfMaps,sourceKey=name,sourceValue=asdf
```
  2. create `ServiceBinding` between deployment and `Service` that has above annotation. 
 

### What is the expected behaviour?

`stats.condition` should have an error telling me that I have specified invalid `sourceValue`

### What is the actual behaviour?

operator throws panic

### Service Binding Operator Logs
```
E0520 16:39:21.734840       1 runtime.go:78] Observed a panic: &reflect.ValueError{Method:"reflect.Value.Interface", Kind:0x0} (reflect: call of reflect.Value.Interface on zero Value)
goroutine 346 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1538ee0, 0xc000573ed8)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x86
panic(0x1538ee0, 0xc000573ed8)
	/usr/lib/golang/src/runtime/panic.go:965 +0x1b9
reflect.valueInterface(0x0, 0x0, 0x0, 0xc000355c01, 0x9, 0x7f3f88777f18)
	/usr/lib/golang/src/reflect/value.go:1021 +0x1ad
reflect.Value.Interface(...)
	/usr/lib/golang/src/reflect/value.go:1016
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0xc0008c99a0, 0x15, 0x1906688, 0xc000a8ebd0, 0x18f4b40, 0xc00007b900, 0x14fa440, 0xc0009a3950, 0x98, 0x0, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:192 +0x44e
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0x0, 0x0, 0x1906688, 0xc000a8ebd0, 0x18f4b40, 0xc00007b900, 0x14fa440, 0xc0009a3930, 0x98, 0x1544c60, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:185 +0x7a5
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.BindingItems(0x1906688, 0xc000a8ebd0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:87 +0x390
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline.HandlerFunc.Handle(0x17a9130, 0x1906688, 0xc000a8ebd0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/api.go:160 +0x3a
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder.(*impl).Process(0xc0007f5710, 0xc0009ee1c0, 0xb, 0xc0004e3020, 0x2)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder/pipeline.go:26 +0xd8
github.com/redhat-developer/service-binding-operator/controllers.(*ServiceBindingReconciler).Reconcile(0xc000518000, 0xc00048dc50, 0xa, 0xc00048dc34, 0xa, 0xc0009a31f0, 0xc000229170, 0xc0004feb48, 0xc0004feb40)
	/tmp/github.com/redhat-developer/service-binding-operator/controllers/servicebinding_controller.go:88 +0x5d8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00016d3b0, 0x159c2c0, 0xc00003bac0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:244 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00016d3b0, 0xc0003b9800)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:218 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(...)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:197
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000523d00)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000523d00, 0x18bda00, 0xc0009612c0, 0x1, 0xc000480000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000523d00, 0x3b9aca00, 0x0, 0x1, 0xc000480000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000523d00, 0x3b9aca00, 0xc000480000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:179 +0x3d6
panic: reflect: call of reflect.Value.Interface on zero Value [recovered]
	panic: reflect: call of reflect.Value.Interface on zero Value

goroutine 346 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1538ee0, 0xc000573ed8)
	/usr/lib/golang/src/runtime/panic.go:965 +0x1b9
reflect.valueInterface(0x0, 0x0, 0x0, 0xc000355c01, 0x9, 0x7f3f88777f18)
	/usr/lib/golang/src/reflect/value.go:1021 +0x1ad
reflect.Value.Interface(...)
	/usr/lib/golang/src/reflect/value.go:1016
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0xc0008c99a0, 0x15, 0x1906688, 0xc000a8ebd0, 0x18f4b40, 0xc00007b900, 0x14fa440, 0xc0009a3950, 0x98, 0x0, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:192 +0x44e
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0x0, 0x0, 0x1906688, 0xc000a8ebd0, 0x18f4b40, 0xc00007b900, 0x14fa440, 0xc0009a3930, 0x98, 0x1544c60, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:185 +0x7a5
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.BindingItems(0x1906688, 0xc000a8ebd0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:87 +0x390
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline.HandlerFunc.Handle(0x17a9130, 0x1906688, 0xc000a8ebd0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/api.go:160 +0x3a
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder.(*impl).Process(0xc0007f5710, 0xc0009ee1c0, 0xb, 0xc0004e3020, 0x2)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder/pipeline.go:26 +0xd8
github.com/redhat-developer/service-binding-operator/controllers.(*ServiceBindingReconciler).Reconcile(0xc000518000, 0xc00048dc50, 0xa, 0xc00048dc34, 0xa, 0xc0009a31f0, 0xc000229170, 0xc0004feb48, 0xc0004feb40)
	/tmp/github.com/redhat-developer/service-binding-operator/controllers/servicebinding_controller.go:88 +0x5d8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00016d3b0, 0x159c2c0, 0xc00003bac0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:244 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00016d3b0, 0xc0003b9800)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:218 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(...)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:197
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000523d00)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000523d00, 0x18bda00, 0xc0009612c0, 0x1, 0xc000480000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000523d00, 0x3b9aca00, 0x0, 0x1, 0xc000480000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000523d00, 0x3b9aca00, 0xc000480000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:179 +0x3d6
```

### Additional Information (Screenshots, etc)

We've seen this happen in the Developer Sandbox with SBO 0.9.1:
```
E0906 14:42:33.026224       1 runtime.go:78] Observed a panic: &reflect.ValueError{Method:"reflect.Value.Interface", Kind:0x0} (reflect: call of reflect.Value.Interface on zero Value)
goroutine 472 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x16125a0, 0xc000c8e940)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x16125a0, 0xc000c8e940)
	/usr/lib/golang/src/runtime/panic.go:969 +0x1b9
reflect.valueInterface(0x0, 0x0, 0x0, 0xc000900101, 0x10, 0xc000048000)
	/usr/lib/golang/src/reflect/value.go:1021 +0x1ad
reflect.Value.Interface(...)
	/usr/lib/golang/src/reflect/value.go:1016
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0x0, 0x0, 0x1a0bb20, 0xc00001c540, 0x19f5c40, 0xc00037e9c0, 0x15c9d20, 0xc000c9e500, 0x98, 0x0, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:259 +0x44c
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.BindingItems(0x1a0bb20, 0xc00001c540)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:90 +0x376
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline.HandlerFunc.Handle(0x18adc80, 0x1a0bb20, 0xc00001c540)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/api.go:168 +0x3a
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder.(*impl).Process(0xc00022ab40, 0x17ccea0, 0xc000847200, 0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder/pipeline.go:25 +0xd8
github.com/redhat-developer/service-binding-operator/controllers.(*BindingReconciler).Reconcile(0xc000083e60, 0xc000498e20, 0x11, 0xc000498d00, 0x17, 0xc000637f00, 0xc0008182d0, 0xc000542128, 0xc000542120)
	/tmp/github.com/redhat-developer/service-binding-operator/controllers/common.go:95 +0x622
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000156510, 0x168d9c0, 0xc00093c120, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:244 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000156510, 0xc0006d2000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:218 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000156510)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:197 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000124100)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000124100, 0x19b5560, 0xc0006c6030, 0x1, 0xc0006d20c0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000124100, 0x3b9aca00, 0x0, 0x100000000000001, 0xc0006d20c0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000124100, 0x3b9aca00, 0xc0006d20c0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:179 +0x416
panic: reflect: call of reflect.Value.Interface on zero Value [recovered]
	panic: reflect: call of reflect.Value.Interface on zero Value

goroutine 472 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x16125a0, 0xc000c8e940)
	/usr/lib/golang/src/runtime/panic.go:969 +0x1b9
reflect.valueInterface(0x0, 0x0, 0x0, 0xc000900101, 0x10, 0xc000048000)
	/usr/lib/golang/src/reflect/value.go:1021 +0x1ad
reflect.Value.Interface(...)
	/usr/lib/golang/src/reflect/value.go:1016
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.collectItems(0x0, 0x0, 0x1a0bb20, 0xc00001c540, 0x19f5c40, 0xc00037e9c0, 0x15c9d20, 0xc000c9e500, 0x98, 0x0, ...)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:259 +0x44c
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect.BindingItems(0x1a0bb20, 0xc00001c540)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect/impl.go:90 +0x376
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline.HandlerFunc.Handle(0x18adc80, 0x1a0bb20, 0xc00001c540)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/api.go:168 +0x3a
github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder.(*impl).Process(0xc00022ab40, 0x17ccea0, 0xc000847200, 0x0, 0x0, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder/pipeline.go:25 +0xd8
github.com/redhat-developer/service-binding-operator/controllers.(*BindingReconciler).Reconcile(0xc000083e60, 0xc000498e20, 0x11, 0xc000498d00, 0x17, 0xc000637f00, 0xc0008182d0, 0xc000542128, 0xc000542120)
	/tmp/github.com/redhat-developer/service-binding-operator/controllers/common.go:95 +0x622
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000156510, 0x168d9c0, 0xc00093c120, 0x0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:244 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000156510, 0xc0006d2000)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:218 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc000156510)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:197 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000124100)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000124100, 0x19b5560, 0xc0006c6030, 0x1, 0xc0006d20c0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000124100, 0x3b9aca00, 0x0, 0x100000000000001, 0xc0006d20c0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000124100, 0x3b9aca00, 0xc0006d20c0)
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/tmp/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:179 +0x416
```

--- a/pkg/reconcile/pipeline/handler/collect/impl.go
+++ b/pkg/reconcile/pipeline/handler/collect/impl.go
@@ -18,6 +18,7 @@ import (
 )
 
 var DataNotMap = errors.New("Returned data are not a map, skip collecting")
+var ErrorValueNotFound = errors.New("Value not found in map")
 
 const (
 	ErrorReadingServicesReason   = "ErrorReadingServices"
@@ -25,6 +26,8 @@ const (
 	ErrorReadingDescriptorReason = "ErrorReadingDescriptor"
 	ErrorReadingBindingReason    = "ErrorReadingBinding"
 	ErrorReadingSecret           = "ErrorReadingSecret"
+
+	ValueNotFound = "ValueNotFound"
 )
 
 func PreFlight(ctx pipeline.Context) {
@@ -249,7 +252,17 @@ func collectItems(prefix string, ctx pipeline.Context, service pipeline.Service,
 			p = ""
 		}
 		for _, n := range v.MapKeys() {
-			collectItems(p, ctx, service, n, v.MapIndex(n).Interface())
+			if mapVal := v.MapIndex(n).Interface(); mapVal != nil {
+				collectItems(p, ctx, service, n, mapVal)
+			} else {
+				condition := apis.Conditions().NotCollectionReady().
+					Msg(fmt.Sprintf("Value for key %v_%v not found", prefix+k.String(), n.String())).
+					Reason(ValueNotFound).Build()
+				ctx.SetCondition(condition)
+				ctx.Error(ErrorValueNotFound)
+				ctx.StopProcessing()
+				return
+			}
 		}
 	case reflect.Slice:
 		for i := 0; i < v.Len(); i++ {

--- a/pkg/reconcile/pipeline/handler/collect/impl_test.go
+++ b/pkg/reconcile/pipeline/handler/collect/impl_test.go
@@ -388,6 +388,38 @@ var _ = Describe("Collect Binding Data", func() {
 			shouldRetry(pipeline.HandlerFunc(collect.BindingItems), collect.ErrorReadingBindingReason, err)
 		})
 
+		Context("on nil value in collected bindings", func() {
+			It("should set an error condition and stop processing the pipeline",
+				func() {
+					serviceResource := &unstructured.Unstructured{}
+
+					ctx := mocks.NewMockContext(mockCtrl)
+					service := mocks.NewMockService(mockCtrl)
+					definition := bindingmocks.NewMockDefinition(mockCtrl)
+					value := bindingmocks.NewMockValue(mockCtrl)
+
+					ctx.EXPECT().Services().Return([]pipeline.Service{service}, nil)
+
+					bindingDefs := []binding.Definition{definition}
+					service.EXPECT().BindingDefs().Return(bindingDefs)
+					service.EXPECT().Resource().Return(serviceResource)
+
+					definition.EXPECT().Apply(serviceResource).Return(value, nil)
+					value.EXPECT().Get().Return(map[string]map[string]interface{}{"java-maven_port": {"foo": nil}})
+
+					ctx.EXPECT().SetCondition(
+						apis.Conditions().NotCollectionReady().
+							Reason(collect.ValueNotFound).
+							Msg("Value for key java-maven_port_foo not found").
+							Build())
+					ctx.EXPECT().Error(collect.ErrorValueNotFound)
+					ctx.EXPECT().StopProcessing()
+
+					collect.BindingItems(ctx)
+				},
+			)
+		})
+
 		Context("on returning unexpected data", func() {
 			var (
 				service *mocks.MockService

--- a/test/acceptance/features/bindAppToServiceAnnotations.feature
+++ b/test/acceptance/features/bindAppToServiceAnnotations.feature
@@ -542,3 +542,45 @@ Feature: Bind an application to a service using annotations
             """
         Then Service Binding "binding-app-via-gvk" is ready
         And The application env var "BACKEND_HOST_INTERNAL_DB" has value "internal.db.stable.example.com"
+
+    Scenario: Application cannot be bound to service containing annotation with an invalid sourceValue value
+        Given Generic test application is running
+        And CustomResourceDefinition backends.stable.example.com is available
+        And The Custom Resource is present
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: $scenario_id
+                annotations:
+                    service.binding/webarrows: path={.spec.connections},elementType=sliceOfMaps,sourceKey=type,sourceValue=asdf
+            spec:
+                connections:
+                  - type: primary
+                    url: primary.example.com
+            status:
+                somestatus: good
+            """
+        When Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: $scenario_id
+            spec:
+                bindAsFiles: false
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: $scenario_id
+                application:
+                    name: $scenario_id
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        Then Service Binding CollectionReady.status is "False"
+        And Service Binding CollectionReady.reason is "ValueNotFound"
+        And Service Binding CollectionReady.message is "Value for key webarrows_primary not found"
+

--- a/test/acceptance/resources/backend_crd.yaml
+++ b/test/acceptance/resources/backend_crd.yaml
@@ -44,6 +44,15 @@ spec:
                       type: string
                     environment:
                       type: string
+                connections:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      type:
+                        type: string
+                      url:
+                        type: string
             status:
               type: object
               properties:

